internal/fs/fs.go

```diff
@@ -2083,17 +2084,15 @@ func (fs *fileSystem) WriteFile(
 	// Serve the request.
 	if fs.isWriteBufferEnabled(in) {
 		// Trigger write file buffering flow.
-		err := in.WriteToBuffer(fs.mountConfig.WriteConfig.BufferSizeMB, op.Data, op.Offset)
-		if err != nil {
-			return err
+		err = in.WriteToBuffer(fs.mountConfig.WriteConfig.BufferSizeMB, op.Data, op.Offset)
+		if err != nil && err.Error() == buffer.NonSequentialWriteError {
+			// Trigger the temp file flow in case of non-sequential writes.
+			err = in.Write(ctx, op.Data, op.Offset)
 		}
 	} else {
 		// Trigger the temp file flow.
-		if err := in.Write(ctx, op.Data, op.Offset); err != nil {
-			return err
-		}
+		err = in.Write(ctx, op.Data, op.Offset)
 	}
-
 	return
 }
```

Comment:
We could write this in a slightly more compressed and perhaps intuitive way: 

if (fs.isWriteBufferEnabled(in) { 
         err := in.WriteToBuffer(fs.mountConfig.WriteConfig.BufferSizeMB, op.Data, op.Offset)
         if (err == nil) 
               return
        log error as warning
        if (err.Error() != NonSequentialWriteError) 
              return
         else log info/warning that we are falling back to file writing
}
err = in.Write(ctx, op.Data, op.Offset)
return

Reply:
Made the suggested change, thanks.